PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/sparkserverless/serverexplore/ui/SparkServerlessProvisionDialog.java

```diff
@@ -0,0 +1,206 @@
+import com.microsoft.azuretools.azurecommons.helpers.NotNull;
+import com.microsoft.intellij.rxjava.IdeaSchedulers;
+import org.apache.commons.lang3.math.NumberUtils;
+import org.jetbrains.annotations.Nullable;
+
+import javax.swing.*;
+import java.awt.event.FocusAdapter;
+import java.awt.event.FocusEvent;
+import java.util.Arrays;
+
+public class SparkServerlessProvisionDialog extends DialogWrapper
+        implements SettableControl<SparkServerlessClusterProvisionSettingsModel> {
+    @NotNull
+    private SparkServerlessClusterProvisionCtrlProvider ctrlProvider;
+
+    @NotNull
+    private SparkServerlessADLAccountNode adlAccountNode;
+
+    @NotNull
+    private AzureSparkServerlessAccount account;
+
+    private JTextField clusterNameField;
+    private JTextField adlAccountField;
+    private JTextField userStorageAccountField;
+    private JTextField previousSparkEventsField;
+    private JTextField masterCoresField;
+    private JTextField masterMemoryField;
+    private JTextField workerCoresField;
+    private JTextField workerMemoryField;
+    private JTextField workerNumberOfContainersField;
+    private JTextField availableAUField;
+    private JTextField totalAUField;
+    private JTextField calculatedAUField;
+    private JLabel masterMemoryLabel;
+    private JLabel masterCoresLabel;
+    private JLabel clusterNameLabel;
+    private JLabel adlAccountLabel;
+    private JLabel previousSparkEventsLabel;
+    private JLabel availableAULabel;
+    private JLabel calculatedAULabel;
+    private JLabel workerCoresLabel;
+    private JLabel workerMemoryLabel;
+    private JLabel workerNumberOfContainersLabel;
+    private JTextField errorMessageField;
+    private JPanel provisionDialogPanel;
+    private JButton refreshButton;
+    private JLabel userStorageAccountLabel;
+
+    public SparkServerlessProvisionDialog(@NotNull SparkServerlessADLAccountNode adlAccountNode,
+                                          @NotNull AzureSparkServerlessAccount account) {
+        // TODO: canBeParent
+        super((Project)adlAccountNode.getProject(), true);
+
+        this.ctrlProvider = new SparkServerlessClusterProvisionCtrlProvider(
+                this, new IdeaSchedulers((Project)adlAccountNode.getProject()), account);
+        this.adlAccountNode = adlAccountNode;
+        this.account = account;
+
+        init();
+        this.setTitle("Provision Spark Cluster");
+        errorMessageField.setBackground(this.provisionDialogPanel.getBackground());
+        errorMessageField.setBorder(BorderFactory.createEmptyBorder());
+        this.setModal(true);
+
+        adlAccountField.addFocusListener(new FocusAdapter() {
+            @Override
+            public void focusLost(FocusEvent e) {
+                // TODO: check validity of ADL account
+                availableAUField.setText(String.valueOf(ctrlProvider.getAvailableAU(adlAccountField.getText())));
```

Comment:
Should be changed to as OK button validation, here only invoking the provider methods.